plugin/storage/memory/memory.go

```diff
@@ -339,3 +331,68 @@ func flattenTags(span *model.Span) model.KeyValues {
 	}
 	return retMe
 }
+
+func isLeaf(span *model.Span, spans []*model.Span) bool {
+	for _, s := range spans {
+		for _, ref := range s.References {
+			if ref.RefType == model.ChildOf && ref.SpanID == span.SpanID {
+				return false // Not a leaf if it's the parent of another span
+			}
+		}
+	}
+	return true
+}
+
+func updateServiceDependencyLinks(parentService, childService string, deps map[string]*model.DependencyLink) {
+	depKey := parentService + "&&&" + childService
+	if _, ok := deps[depKey]; !ok {
+		deps[depKey] = &model.DependencyLink{
+			Parent:    parentService,
+			Child:     childService,
+			CallCount: 1,
+		}
+	} else {
+		deps[depKey].CallCount++
+	}
+}
+
+func inferServiceName(span *model.Span, isLeaf bool) string {
+	serviceName := "inferred"
+
+	if isLeaf {
```

Comment:
when would this be false?

Reply:
I've reconsidered the isLeaf parameter and agree that it's not needed right now since we only use inferServiceName for leaf spans. Keeping the code simple and avoiding unnecessary complexity is a priority, so I removed the isLeaf check. This will make the function more straightforward for our current needs.